cpp/util/ManagedArray.h

```diff
@@ -96,8 +98,14 @@ template<typename T> class ManagedArray
                 (*m_size) *= (*m_shape)[i];
             }
 
-            m_data = std::shared_ptr<std::shared_ptr<T>>(
-                new std::shared_ptr<T>(new T[size()], std::default_delete<T[]>()));
+            // We make use of C-style arrays here rather than any alternative
+            // because we need the underlying data representation to be
+            // compatible with numpy on the Python side. We _could_ do this
+            // with a different data structure like std::vector, but it would
```

Comment:
Or `std::span`, coming soon to a C++20 near you. https://en.cppreference.com/w/cpp/container/span

Multi-dimensional spans have been proposed as well.

Reply:
And by coming soon, you mean coming to your compiler of choice by... 2026. Maybe 2023, they are getting faster with updating to standards haha.